File: app/src/app/core/models/completion.ts
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

export type Completion = {
  cmid: number;
  module: string;
  instance: number;
  state: number | null;
  completedTs: number | null;
  tracking: number;
  overrideBy: number | null;
  isEnabled: boolean;
  isValueUsed: boolean;
  isAutomatic: boolean;
  isTrackedUser: boolean;
  isUserVisible: boolean;
  isOverallComplete: boolean;
};

File: api/src/main/kotlin/es/ubu/lsi/ubumonitorweb/domain/Completion.kt
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

package es.ubu.lsi.ubumonitorweb.domain

data class Completion(
  val cmid: Int,
  val module: String,
  val instance: Int,
  val state: Int?,
  val completedTs: Long?,
  val tracking: Int,
  val overrideBy: Int?,
  val isEnabled: Boolean,
  val isValueUsed: Boolean,
  val isAutomatic: Boolean,
  val isTrackedUser: Boolean,
  val isUserVisible: Boolean,
  val isOverallComplete: Boolean,
)
